src/download.test.ts

```diff
@@ -131,6 +131,12 @@ describe("getSkylinkUrl", () => {
     expect(url).toEqual(`${expectedUrl}?no-response-metadata=true`);
   });
 
+  it("should return correctly formed URLs with no-response-metadata set and with forced download", () => {
+    const url = client.getSkylinkUrl(skylink, { download: true, noResponseMetadata: true });
+
+    expect(url).toEqual(`${expectedUrl}?attachment=true&no-response-metadata=true`);
```

Comment:
I think these might appear in a different order and randomly fail the test. IIRC, ordering is not guaranteed when serializing objects in JS and this is what essentially happens in `addUrlQuery`:
```js
  query = { ...parsed.query, ...query };
  parsed.set("query", query);
  return parsed.toString();
```
I might be wrong, though.

In any case, not a blocker.